CRM/Event/Form/Registration/Confirm.php

```diff
@@ -583,16 +583,6 @@ public function postProcess() {
           if (empty($value['email'])) {
             $value['email'] = CRM_Utils_Array::valueByRegexKey('/^email-/', $value);
           }
-
-          if (is_object($payment)) {
-            // Not quite sure why we don't just user $value since it contains the data
-            // from result
-            // @todo ditch $result & retest.
-            list($result, $value) = $this->processPayment($payment, $value);
-          }
-          else {
-            CRM_Core_Error::fatal($paymentObjError);
-          }
```

Comment:
Can we clear up `$result` before moving the code that actually takes the contribution.  `$result` which is passed into processContribution is set here: https://github.com/civicrm/civicrm-core/blob/e1cf69bdcbbbd73136744b75550e3ccf098da3a7/CRM/Event/Form/Registration/Confirm.php#L563 and then we have a different `$result` below which is the result of doPayment.  Maybe just remove that line and set $result in the call to processContribution to NULL directly?

Reply:
@mattwire yes I agree! fixed